Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/SolidColorIconEngine.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <QPainter>
+
+#include <EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/SolidColorIconEngine.h>
+
+namespace EMStudio
+{
+    SolidColorIconEngine::SolidColorIconEngine(const QColor& color)
+        : QIconEngine()
+    {
+        setColor(color);
+    }
+
+    SolidColorIconEngine::SolidColorIconEngine(const AZ::Color& color)
+        : QIconEngine()
+    {
+        setColor(color);
+    }
+
+    QColor SolidColorIconEngine::color() const
+    {
+        return m_color;
+    }
+
+    void SolidColorIconEngine::setColor(const QColor& color)
+    {
+        m_color = color;
+    }
+
+    void SolidColorIconEngine::setColor(const AZ::Color& color)
+    {
+        m_color = QColor::fromRgb(color.GetR8(), color.GetG8(), color.GetB8(), color.GetA8());
+    }
+
+    void SolidColorIconEngine::paint(QPainter* painter, const QRect& rect, QIcon::Mode mode, QIcon::State state)
+    {
+        AZ_UNUSED(mode);
+        AZ_UNUSED(state);
+
+        painter->setBrush(m_color);
+        painter->drawRect(rect);
+    }
+
+    QIconEngine* SolidColorIconEngine::clone() const
```

Comment:
update: const broken, please drop `const` or return a `const` pointer (or have both and overload) - see https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/const.md#classes-of-const-in_correctness for a good overview on this

Reply:
Actually in this case as it's a clone function this is okay sorry, but it can be a problem in other cases, so this particular instance can stay as is as we're returning a new object (we maybe want to use a unique pointer here though to show the transfer of ownership which would be clearer to the caller rather than a raw pointer)

Reply:
This is an override of a Qt API, so we can't return a unique_ptr.

Reply:
Aah I see, cool thanks for clarifying